### PR TITLE
Allow dockerfile patterns without a trailing newline

### LINKED ISSUE #4771: Assume newline at the end of dockerfile patterns
**Describe the bug**

Something removes the newlines from dockerfile patterns when using the Playground. The pattern parser then fails to accept it as valid.

Instead, we should accept patterns that don't end with a newline.

See https://semgrep.dev/s/mjambon:dockerfile-missing-newline

image

A workaround is to manually hit Enter at the end of the pattern(s), every time.

### PR DESCRIPTION
**Pull request by mjambon**
Fixes #4771

PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: semgrep-core/tests/dockerfile/no-trailing-newline.dockerfile [hunk 1]
```diff
@@ -0,0 +1 @@
+CMD hello
```
Comment (aryx) [inline semgrep-core/tests/dockerfile/no-trailing-newline.dockerfile:2]: hmm you miss a #ERROR: match here I think

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
What is your "Test plan:"? 
Haha! Got you :)

**Comment by aryx** [comment]
CI error seems unrelated. github flakyness?
```
ERROR    semgrep.output:output.py:245 Failed to download config from https://semgrep.dev/r/python.lang.correctness.useless-eqeq.useless-eqeq: bad status code:
503 returned by config url: https://semgrep.dev/r/python.lang.correctness.useless-eqeq.useless-eqeq
VERBOSE  semgrep.semgrep_main:semgrep_main.py:326 running 0 rules from 0 config  (1 config files were invalid)
ERROR    semgrep.output:output.py:245 invalid configuration file found (1 configs were invalid)
=============================== warnings summary ===============================
semgrep/ignores.py:104
semgrep/ignores.py:104
  /home/runner/work/semgrep/semgrep/semgrep/semgrep/ignores.py:104: DeprecationWarning: invalid escape sequence \:
    """

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/e2e/test_login.py::test_login - assert 1 == 0
```

**Comment by aryx** [comment]
or semgrep registry flakyness.

File: semgrep-core/tests/dockerfile/no-trailing-newline.dockerfile (hunk 1 above)
Comment (mjambon) [reply semgrep-core/tests/dockerfile/no-trailing-newline.dockerfile:2]: thanks for the fix!

**Comment by mjambon** [comment]
> What is your "Test plan:"?

It's `make test` unless otherwise specified (yes, I should have run it manually before committing).

**Pull request merged**